opentelemetry-sdk/Cargo.toml

```diff
@@ -29,6 +29,7 @@ tokio = { workspace = true, features = ["rt", "time"], optional = true }
 tokio-stream = { workspace = true, optional = true }
 http = { workspace = true, optional = true }
 tracing = {workspace = true, optional = true}
+rustc-hash = "2.0.0"
```

Comment:
I should probably try other hash functions, but results that I posted is with this crate.
Generally, my idea was, that if it is good enough for rustc compiler (which operates on a lot of small strings), it should be perfect fit for attribute sets too.
But if you want, I might as well test with other crates.

I remember that @cijothomas pointed me to this comment:
https://github.com/open-telemetry/opentelemetry-rust/pull/1564/files#diff-efe82800c0ca89e5cdf4c14f0674f753b0148351f0bfad79f89b20d295c4e6e4R58
but I'm a bit afraid that attribute-sets in current benchmarks/stress-test is different from real world usage (using  opentelemetry-semantic-conventions)...

Reply:
In general, when running benchmarks to compare performance for metric updates, please run the benchmarks in [`metrics_counter.rs`](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/benches/metrics_counter.rs) instead of [`metric.rs`](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/benches/metric.rs). We haven't really fine-tuned the benchmarks in `metrics.rs`. We might also remove if it doesn't help that much.